### ISSUE TITLE
IL gen is not protected against errors originating in lowering

### ISSUE THREAD

**Issue by TyOverby** [dotnet/roslyn#20085]
In MethodCompiler.GetEntryPoint, we lower without checking the lowered form for errors.  Right now it is impossible to lower and get errors, but this should be future-proofed.

**Comment by jaredpar** [comment]
I believe the philosophy here is that lowering should never produce errors. Any error that could be generated by lowering should be caught in earlier compiler phases. 

It may make sense though to guard against this. For instance if lowering did produce an error, fail the compilation with the equivalent of the native "internal error occurred". 

Thoughts? @AlekseyTs, @gafter

**Comment by AlekseyTs** [comment]
> I believe the philosophy here is that lowering should never produce errors. Any error that could be generated by lowering should be caught in earlier compiler phases.

This is not correct. Lowering can produce errors and it does in many cases. Ty opened that issue based on my CR feedback.
